frmts/heif/heifdataset.cpp

```diff
@@ -637,14 +655,92 @@ GDALHEIFRasterBand::GDALHEIFRasterBand(GDALHEIFDataset *poDSIn, int nBandIn)
                                         "IMAGE_STRUCTURE");
     }
 #endif
+
+#ifdef LIBHEIF_SUPPORTS_TILES
+    nBlockXSize = poDSIn->m_tiling.tile_width;
```

Comment:
the 0x0 block size error must come from m_tiling.tile_width/tile_heigh being 0. Perhaps that's the way libheif signals a non-tiled dataset? In which case you might want to go to to the scanline oriented approach below and potentially use the "old" IReadBlock() way too ? Just guessing, having not checked how libheif deals with tiling...

Reply:
Its is supposed to be uniform handling (i.e. non-tiled is a single tile).

The problem was in the overview data set - I hadn't added tiling for that. Should now be fixed.